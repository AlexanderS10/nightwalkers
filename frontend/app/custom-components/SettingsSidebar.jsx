import React from "react";
import {
  UserRound,
  Flag,
  MapPinCheck,
  ShieldPlus,
  Settings,
  LogOut,
  X,
} from "lucide-react";
import { Avatar, AvatarFallback, AvatarImage } from "@/components/ui/avatar";
import {
  Sidebar,
  SidebarContent,
  SidebarGroup,
  SidebarGroupContent,
  SidebarGroupLabel,
  SidebarMenu,
  SidebarMenuButton,
  SidebarMenuItem,
  SidebarSeparator,
  useSidebar,
} from "@/components/ui/sidebar";

const items = [
  {
    title: "Profile",
    url: "#",
    icon: UserRound,
  },
  {
    title: "Routes",
    url: "#",
    icon: MapPinCheck,
  },
  {
    title: "Report",
    url: "#",
    icon: Flag,
  },
  {
    title: "Resources",
    url: "#",
    icon: ShieldPlus,
  },
  {
    title: "Settings",
    url: "#",
    icon: Settings,
  },
  {
    title: "Sign Out",
    url: "#",
    icon: LogOut,
  },
];
const SettingsSidebar = () => {
  const {
    state,
    open,
    setOpen,
    openMobile,
    setOpenMobile,
    isMobile,
    toggleSidebar,
  } = useSidebar();

  return (
    <Sidebar side="right" collapsible="offcanvas">
      <SidebarContent>
        {/* Banner */}
        <SidebarGroup>
          <SidebarGroupContent>
            <div className="w-full">
              <div className="flex justify-between">
                <div className="flex-1" /> {/* Spacer */}
                <div className="p-2 cursor-pointer hover:opacity-70">
                  <X
                    onClick={() =>
                      isMobile ? setOpenMobile(false) : setOpen(false)
                    }
                  />
                </div>
              </div>
              <div className="flex flex-col items-center gap-5">
                <Avatar className="h-24 w-24">
                  <AvatarImage
                    src="https://github.com/shadcn.png"
                    alt="@shadcn"
                  />
                  <AvatarFallback>CN</AvatarFallback>
                </Avatar>
                <span>Username</span>
              </div>
            </div>
          </SidebarGroupContent>
        </SidebarGroup>
        <SidebarSeparator />
        {/* Links */}
        <SidebarGroup>
          <SidebarGroupLabel>Application</SidebarGroupLabel>
          <SidebarGroupContent>
            <SidebarMenu>
              {items.map((item) => (
                <SidebarMenuItem key={item.title}>
                  <SidebarMenuButton asChild>
                    <a href={item.url}>
                      <item.icon />
                      <span>{item.title}</span>
                    </a>
                  </SidebarMenuButton>
                  <SidebarSeparator />
                </SidebarMenuItem>
              ))}
            </SidebarMenu>
          </SidebarGroupContent>
        </SidebarGroup>
      </SidebarContent>
    </Sidebar>
  );
};

export default SettingsSidebar;
