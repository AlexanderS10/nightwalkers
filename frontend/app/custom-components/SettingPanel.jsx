import React, { useEffect, useState } from "react";
import SettingsSidebar from "./SettingsSidebar";
import { SidebarProvider, SidebarTrigger } from "@/components/ui/sidebar";

const SettingPanel = () => {
  const [open, setOpen] = useState(false);
  useEffect(() => {
    console.log(open)
  }, [open]);
  return (
    <SidebarProvider
      open={open}
      onOpenChange={setOpen}
    >
      <SidebarTrigger />
      <SettingsSidebar />
    </SidebarProvider>
  );
};

export default SettingPanel;
