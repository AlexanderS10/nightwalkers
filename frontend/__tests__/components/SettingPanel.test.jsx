import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import SettingPanel from "@/app/custom-components/SettingPanel";
import SettingsSidebar from "@/app/custom-components/SettingsSidebar";

// jest.mock("@/components/ui/sidebar", () => ({
//   SidebarProvider: ({ children }) => (
//     <div data-testid="sidebar-provider">{children}</div>
//   ),
//   Sidebar: ({ children }) => <div role="complementary">{children}</div>,
//   SidebarContent: ({ children }) => <div>{children}</div>,
//   useSidebar: () => ({
//     open: false,
//     toggleSidebar: jest.fn(),
//   }),
// }));

// // Mock the next/link component
// jest.mock("next/link", () => {
//   return ({ children, href }) => {
//     return <a href={href}>{children}</a>;
//   };
// });

// describe("SettingPanel", () => {
//   it("renders without crashing", () => {
//     render(<SettingPanel />);
//     expect(screen.getByRole("complementary")).toBeInTheDocument();
//   });

//   it("toggles sidebar visibility when trigger is clicked", () => {
//     render(<SettingPanel />);
//     const trigger = screen.getByRole("button");
//     fireEvent.click(trigger);
//     // Check if sidebar is visible
//     expect(screen.getByRole("complementary")).toBeInTheDocument();
//   });
// });

// describe("SettingsSidebar", () => {
//   it("renders user avatar and username", () => {
//     render(<SettingsSidebar />);
//     expect(screen.getByText("Username")).toBeInTheDocument();
//   });

//   it("renders all navigation items", () => {
//     render(<SettingsSidebar />);
//     const navigationItems = [
//       "Profile",
//       "Routes",
//       "Report",
//       "Resources",
//       "Settings",
//       "Sign Out",
//     ];

//     navigationItems.forEach((item) => {
//       expect(screen.getByText(item)).toBeInTheDocument();
//     });
//   });

//   it("renders correct navigation links", () => {
//     render(<SettingsSidebar />);
//     expect(screen.getByRole("link", { name: /profile/i })).toHaveAttribute(
//       "href",
//       "/users/settings/profile"
//     );
//     expect(screen.getByRole("link", { name: /routes/i })).toHaveAttribute(
//       "href",
//       "/users/settings/routes"
//     );
//     expect(screen.getByRole("link", { name: /report/i })).toHaveAttribute(
//       "href",
//       "/users/settings/report"
//     );
//   });
// });
