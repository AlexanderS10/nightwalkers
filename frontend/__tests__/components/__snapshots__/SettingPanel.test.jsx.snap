// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingPanel matches snapshot 1`] = `
<div>
  <div
    data-open="false"
    data-testid="sidebar-provider"
  >
    <button
      data-testid="sidebar-trigger"
    >
      Toggle
    </button>
    <div
      data-testid="settings-sidebar"
    >
      Sidebar Content
    </div>
  </div>
</div>
`;
